### ISSUE TITLE
Error registering classes Basemanager from multiprocessing.managers

### ISSUE THREAD

**Issue by Stigjb** [python/typeshed#2356]
This snippet from the documentation to the [`multiprocessing`](https://docs.python.org/3.6/library/multiprocessing.html#customized-managers) module raises three errors in `mypy`.

```
from multiprocessing.managers import BaseManager

class MathsClass:
    def add(self, x, y):
        return x + y

    def mul(self, x, y):
        return x * y

class MyManager(BaseManager):
    pass

MyManager.register('Maths', MathsClass)

if __name__ == '__main__':
    with MyManager() as manager:
        maths = manager.Maths()
        print(maths.add(4, 3))
        print(maths.mul(7, 8))
```

The errors are:
```
error:Argument 1 to "register" of "BaseManager" has incompatible type "str"; expected "BaseManager"
error:Argument 2 to "register" of "BaseManager" has incompatible type "Type[MathsClass]"; expected "str"
error:"BaseManager" has no attribute "Maths"
```
Here is the method signature in typeshed:
https://github.com/python/typeshed/blob/4ca0a63027e01698313284c29c49a0dc611efeff/stdlib/3/multiprocessing/managers.pyi#L23

`register` is a classmethod in BaseManager, maybe that matters.

**Comment by JelleZijlstra** [comment]
Sounds like we're missing an `@classmethod` decorator. Would you mind submitting a PR to fix that?

As for the last error, it's unlikely that you'll be able to convince a type checker that the `manager.Maths` attribute exists; it's just too dynamic.

**Comment by Stigjb** [comment]
fixed in #2357
